supply/lib/supply/uploader.rb

```diff
@@ -105,13 +117,23 @@ def upload_binary
             upload_changelog(language, apk_version_code)
           end
         end
-
       else
         Helper.log.info "No apk file found, you can pass the path to your apk using the `apk` option"
       end
+
+      UI.user_error!("Could not upload #{apk_path}") unless apk_version_code
```

Comment:
Moved the error up into the if branch. I think it's actually better here because it's raised the second we know it failed instead of trying to upload obbs and changelogs with the wrong `apk_version_code`
```
Helper.log.info "Preparing apk at path '#{apk_path}' for upload..."
apk_version_code = client.upload_apk(apk_path)
UI.user_error!("Could not upload #{apk_path}") unless apk_version_code
```